src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -303,6 +303,39 @@ public StudentExamResource(ExamAccessService examAccessService, StudentExamServi
         return ResponseEntity.ok(testRun);
     }
 
+    /**
+     * GET /courses/{courseId}/exams/{examId}/student-exams/exercise-ids : Find the exercise ids for the current user exam.
+     * Currently this is used for exam notification problem statement updates as an auxiliary method.
+     *
+     * @param courseId  the course to which the student exam belongs to
+     * @param examId    the exam to which the student exam belongs to
+     * @return the ResponseEntity of type Long[] corresponding to an array of the found exercise ids
+     */
+    @GetMapping("/courses/{courseId}/exams/{examId}/student-exams/exercise-ids")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<Long[]> getStudentExamExerciseIds(@PathVariable Long courseId, @PathVariable Long examId) {
```

Comment:
I don't like this REST call. Why do we need it? The exercise ids should be available on the client as soon as the exam has started. There is no need to fetch those from the server.

Please remove this REST call again and try to implement the feature without it!